chapter12/index.md

```diff
@@ -48,39 +48,44 @@ Now let's `SELECT Person.strength;` and see if it works:
 
 Looks like it worked.
 
-So now let's overload the `fight()` function. Right now it only works for one `Person` vs. another `Person`, but in the book all the characters get together to try to defeat Dracula. We'll need to overload the function so that more than one character can work together to fight. There are a lot of ways to do it, but we'll choose a simple one:
+So now let's overload the `fight()` function. Right now it only works for one `Person` vs. another `Person`, but in the book all the characters get together to try to defeat Dracula. We'll need to overload the function so that more than one character can work together to fight.
 
 ```sdl
-function fight(names: str, one: int16, two: str) -> str
+function fight(people_names: array<str>, opponent: Person) -> str
   using (
-    WITH opponent := assert_single((SELECT Person FILTER .name = two))
+    WITH
+        people := (SELECT Person FILTER .name IN array_unpack(people_names)),
     SELECT
-        names ++ ' win!' IF one > opponent.strength ELSE
-        two ++ ' wins!'
+        (array_join(people_names, ', ') ?? 'Fighters') ++ ' win!'
+        IF (sum(people.strength) ?? 0) > (opponent.strength ?? 0)
+        ELSE (opponent.name ?? 'Opponent') ++ ' wins!'
   );
 ```
 
+With this overload, we accept two arguments: an array of the names of the fighters and a `Person` object that is their opponent. You're seeing some new standard library functions in use here that we'll dig into more later. For now, here are the basics you need to know:
+
+- We need to call `array_unpack` when we assign the value to `people` in our `WITH`. This converts our array to a set which is needed because `IN` only works with a set.
+- At the beginning of our `SELECT`, we call `array_join` on `people_names` so that this is displayed as a string. In order to use the coalescing operator (`??`), both operands must be of the same type. Since our alternative is the string `"Fighters"`, we need this side to be a string as well.
```

Comment:
These need to be re-written to account for the updated query using `contains` instead of the messier `array_unpack` and somewhat different handling of the potentially empty array of names.